url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/Profile.java

```diff
@@ -0,0 +1,801 @@
+
+	public String getTemplatePathFor(Request r, String fileName) {
+		// get path from command line
+		
+		// check for GET var "template"
+		if (r.getGetVars().isSet("template"))
+			return buildPath(TEMPLATE_PATH, r.getGetVars().getValue("template"), fileName);
+		
+		// get template name from command line
+		if (templateName != null)
+			return buildPath(TEMPLATE_PATH, templateName, fileName);
+			
+		// just default
+		return buildPath(TEMPLATE_PATH, fileName);
+	}
+
+	/**
+	 * @return True if there are multiple profiles to pick from
+	 */
+	public boolean hasMultipleProfiles() {
+		return possibleProfiles != null && possibleProfiles.length > 1;
+	}
+
+	/**
+	 * @return True if there are multiple users to pick from
+	 */
+	public boolean hasMultipleUsers() {
+		return possibleUsers != null && possibleUsers.size() > 1;
+	}
+	
+	/**
+	 * @param args The ones from main()
+	 */
+	public void initialize(String[] args) {
+		// check for a specified profile
+		ArgumentParser p = ArgumentParser.getInstance();
+		OptionSet o = p.parse(args);
+		
+		if (o.has("?")) {
+			try {
+				p.parser.printHelpOn(System.out);
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			System.exit(0);
+		}
+
+		if (o.has(p.profile)) {
+			File path = p.profile.value(o);
+			if (path.exists() && path.isFile() && path.canRead()) {
+				loadFromFile(path);
+				return;
+			} else {
+				System.err.println("Given profile path does not exist or cannot be read.");
+				System.err.println("Given: " + path.getAbsolutePath());
+				System.exit(1);
+			}
+		}
+		
+		// if sufficient args on command line, use them
+		if (o.has(p.emtype) || o.has(p.dbtype)) {
+			profileFromCommandLine = true;
+			loadFromOptions(o);
+			return;
+		}
+		
+		// look in the profile directory
+		File profileDir = p.profileDir.value(o);
+		File[] possibleProfiles = null;
+		if (profileDir.exists() && profileDir.isDirectory())
+			possibleProfiles = profileDir.listFiles(new FilenameFilter() {
+				@Override
+				public boolean accept(File dir, String name) {
+					return name.endsWith(".profile");
+				}				
+			});
+
+		if (possibleProfiles != null && possibleProfiles.length == 1) {
+			// there's just one; load it!
+			loadFromFile(possibleProfiles[0]);
+			return;
+		} else if (possibleProfiles != null && possibleProfiles.length > 1 ){		
+			// else, we'll prompt for a selection (there are multiple in directory)
+			this.possibleProfiles = possibleProfiles;
+			port = p.portArg.value(o); // get the default, anyway
+			return;
+		}
+		
+		// last case: just use defaults
+		loadFromOptions(o);
+	}
+
+	/**
+	 * @return True if we have a profile selected
+	 */
+	public boolean isSelected() {
+		return id > -1;
+	}
+	
+	/**
+	 * @return True if synchronization is on
+	 */
+	public boolean isSyncing() {
+		return syncSettings != null;
+	}
+	
+	public void loadFromFile(File file) {
+		StringBuffer b = new StringBuffer();
+		try {
+			BufferedReader in = new BufferedReader(new FileReader(file));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *in*. Execution paths that do not contain closure statements were detected. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).